src/main/java/seedu/address/logic/commands/DeleteRatingCommand.java

```diff
@@ -0,0 +1,115 @@
+    private Person editedPerson;
+
+    public DeleteRatingCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() {
+        requireNonNull(targetPerson);
+        requireNonNull(editedPerson);
+
+        try {
+            model.updatePerson(targetPerson, editedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new AssertionError("Deleting a person's rating should not result in a duplicate");
+        } catch (PersonNotFoundException pnfe) {
+            throw new AssertionError("The target person cannot be missing");
+        }
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_DELETE_RATING_SUCCESS, targetPerson.getName()));
+    }
+
+    @Override
+    protected void preprocessUndoableCommand() throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        targetPerson = lastShownList.get(targetIndex.getZeroBased());
+
+        if (targetPerson.getRating().getOverallScore() == Rating.DEFAULT_SCORE) {
+            throw new CommandException(String.format(MESSAGE_PERSON_NOT_RATED, targetPerson.getName()));
+        }
+
+        editedPerson = createEditedPerson(targetPerson);
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code targetPerson}
+     * with rating deleted.
+     */
+    private static Person createEditedPerson(Person targetPerson) {
+        assert targetPerson != null;
+
+        Name name = targetPerson.getName();
+        Phone phone = targetPerson.getPhone();
+        Email email = targetPerson.getEmail();
+        Address address = targetPerson.getAddress();
+        ExpectedGraduationYear expectedGraduationYear = targetPerson.getExpectedGraduationYear();
+        Major major = targetPerson.getMajor();
+        Rating defaultRating = new Rating(Rating.DEFAULT_SCORE, Rating.DEFAULT_SCORE,
+                Rating.DEFAULT_SCORE, Rating.DEFAULT_SCORE);
+        Resume resume = targetPerson.getResume();
+        InterviewDate interviewDate = targetPerson.getInterviewDate();
+        Set<Tag> tags = targetPerson.getTags();
+
+        return new Person(name, phone, email, address,
+                expectedGraduationYear, major, defaultRating, resume, interviewDate, tags);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof DeleteRatingCommand // instanceof handles nulls
+                && this.targetIndex.equals(((DeleteRatingCommand) other).targetIndex) // state check
+                && Objects.equals(this.targetPerson, ((DeleteRatingCommand) other).targetPerson));
```

Comment:
I don't think this equal method will work since `equals` method in `Person` does not compare Rating.

Reply:
It is completely my bad...